cli/cmd/compose/compose.go

```diff
@@ -85,22 +85,26 @@ func Command(contextType string) *cobra.Command {
 
 	command.AddCommand(
 		upCommand(contextType),
-		downCommand(),
-		psCommand(),
-		listCommand(),
-		logsCommand(),
-		convertCommand(),
-		runCommand(),
+		downCommand(composeOpts),
```

Comment:
You'll need to pass the `composeOpts` struct by reference here.

It's confusing but this is where we register commands and flags are only parsed later on. Passing the struct by copy here means that the persistent flags will set the value of the struct declared in this function but the passed struct will be copied before this and never have its fields set.

Reply:
Absolutely - I have no idea why I didn't have the execution order in mind here.